src/main/kotlin/no/skatteetaten/aurora/boober/service/RedeployService.kt

```diff
@@ -32,11 +30,16 @@ class RedeployService(val openShiftClient: OpenShiftClient,
         }
     }
 
-    fun triggerRedeploy(namespace: String, name: String, imageStream: ImageStream?): RedeployResult {
+    fun triggerRedeploy(deploymentConfig: DeploymentConfig, imageStream: ImageStream?): RedeployResult {
+        val namespace = deploymentConfig.metadata.namespace
+        val name = deploymentConfig.metadata.name
+        val imageChangeTriggerTagName = deploymentConfig.findImageChangeTriggerTagName()
+                ?: throw IllegalArgumentException("No imageChangeTriggerName found")
+
         return if (imageStream == null) {
```

Comment:
Jeg lurer på om man i denne sammenhengen kanskje burde gjøre requestDeployment(deploymentConfig) og rolloutDeployment(imageStream, imageChangeTriggerTagName), og at man henter namespace og name fra sine respektive objekter internt i hver metode. Strengt tatt kunne man da hatt overloading (selv om jeg debatterer med meg selv om det er bedre eller ei); redeploy(deploymentConfig) og redeploy(imageStream, imageChangeTriggerTagName). Da blir det muligens litt tydligere at man ønsker oppnå det samme, bare med litt forskjellige mekanismer?

(Selv om det ikke kan oppstå problemer med dette i praksis er det nok formelt sett mest korrekt å bruke name og namespace for ImageStream i rolloutDeployment og for DeploymentConfig i requestDeployment for å garantere at operasjonene blir utført i riktig namespace og mot riktig imagestream/deploymentconfig. I dag er det en implisitt sammenheng, men dersom dette skulle endres, f.eks. ved at man har flere imagestreams pr deploymentconfig (flere containere i samme pod) må man passe på så man bruker riktig.)

Reply:
Enig i at requestDeployment(deploymentConfig) og rolloutDeployment(imageStream, imageChangeTriggerTagName) er tydeligere, jeg gjør om til det.